File: zh/philosophy.c
#include <stdio.h>
#include <string.h>

int main () {
    char question[] = "什么是";
    char idea[]     = "哲学";
    char suffix[]   = "的哲学";
    char symbol[]   = "？";

    char iteration[42] = ""; // assume 42 is infinite ∞

    while(1) {
        printf("%s%s%s%s\n", question, idea, iteration, symbol);
        strcat(iteration, suffix);
    }
}

File: zh/philosophy.js
(function philosophy() {
    var question = '什么是';
    var idea     = '哲学';
    var suffix   = '的哲学';
    var symbol   = '？';

    var iteration = '';

    // assume 42 is infinite ∞
    for (var i = 0; i < 42; i++) {
        var output = question + idea + iteration + symbol;

        console.log(output);

        iteration = iteration + suffix;
    }
})();
